pytest/lib/mocknet.py

```diff
@@ -327,6 +336,103 @@ def stop_node(node):
         m.run('sudo -u ubuntu -i', input=TMUX_STOP_SCRIPT)
 
 
+def create_and_upload_genesis(nodes, genesis_template_filename):
+    mocknet_genesis_filename = '/tmp/mocknet_genesis.json'
```

Comment:
Use `tempfile.NamedTemporaryFile`. You can change `create_genesis_file` to take output file handler as argument and that’ll work nicely.